src/main/java/com/softserve/academy/spaced/repetition/service/impl/CourseServiceImpl.java

```diff
@@ -209,4 +212,18 @@ public Course addDeckToCourse(Long courseId, Long deckId) {
     public List<Course> findAllCoursesBySearch(String searchString) {
         return courseRepository.findByNameIgnoreCaseContainingOrDescriptionIgnoreCaseContaining(searchString, searchString);
     }
+
+    @Override
+    @Transactional
```

Comment:
- lets move this logic to the updateCourse() method. Any way we will be updating the price with the name and other course fields.

- There is a bug in your code. You are using the same course id for both:

1. coursePriceRepository.findOne(courseId)

2. courseRepository.findOne(courseId)

- Please fix Sonar issues introduced you.

Reply:
I could move the logic to updateCourse() method, there's no implementation of updating course on web part of the project and this method is not used.